libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -181,11 +158,42 @@ internal class MapboxTripSession(
         if (state == TripSessionState.STARTED) {
             return
         }
+        setNavigatorObserver()
         tripService.startService()
         startLocationUpdates()
         state = TripSessionState.STARTED
     }
 
+    private fun setNavigatorObserver() {
+        navigator.setNavigatorObserver(
+            NavigatorObserverImpl(
```

Comment:
```suggestion
            NavigatorObserver(
```
I think we can remove `NavigatorObserverImpl ` and `TripStatusObserver`. It will simplify logic. 

```kotlin
navigator.setNavigatorObserver(object : NavigatorObserver {
    override fun onStatus(origin: NavigationStatusOrigin, status: NavigationStatus) {
        jobController.scope.launch {
            val tripStatus = MapboxNativeNavigatorImpl.generateTripStatusFrom(status)
            ....
        }
    }
    }
)
```